__tests__/integration/atomic_swap.test.js

```diff
@@ -0,0 +1,82 @@
+import { GenesisWalletHelper } from './helpers/genesis-wallet.helper';
+import {
+  createTokenHelper,
+  DEFAULT_PIN_CODE,
+  generateWalletHelper,
+  stopAllWallets,
+  waitForTxReceived,
+} from './helpers/wallet.helper';
+import { HATHOR_TOKEN_CONFIG } from '../../src/constants';
+import SendTransaction from '../../src/new/sendTransaction';
+import PartialTxProposal from '../../src/wallet/partialTxProposal';
+import storage from '../../src/storage';
+
+describe('partial tx proposal', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('Should exchange tokens between wallets', async () => {
```

Comment:
It seems to me that it would be important to do a test almost identical to this one, but making a transaction with change outputs on just one or two of the tokens exchanged.

Reply:
Changed the test to include change outputs